packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -15,20 +15,27 @@ import { sidebars } from '../settings-sidebar';
 const { Tabs } = unlock( componentsPrivateApis );
 
 const SettingsHeader = () => {
-	const { documentLabel, isTemplateMode } = useSelect( ( select ) => {
-		const { getPostTypeLabel, getRenderingMode } = select( editorStore );
-
-		return {
-			// translators: Default label for the Document sidebar tab, not selected.
-			documentLabel: getPostTypeLabel() || _x( 'Document', 'noun' ),
-			isTemplateMode: getRenderingMode() === 'template-only',
-		};
-	}, [] );
+	const { documentLabel, isTemplateMode, isPatternMode } = useSelect(
+		( select ) => {
+			const { getPostTypeLabel, getRenderingMode } =
+				select( editorStore );
+			const renderingMode = getRenderingMode();
+			return {
+				// translators: Default label for the Document sidebar tab, not selected.
+				documentLabel: getPostTypeLabel() || _x( 'Document', 'noun' ),
+				isTemplateMode: renderingMode === 'template-only',
+				isPatternMode: renderingMode === 'pattern-only',
+			};
+		},
+		[]
+	);
 
 	return (
 		<Tabs.TabList>
 			<Tabs.Tab tabId={ sidebars.document }>
-				{ isTemplateMode ? __( 'Template' ) : documentLabel }
+				{ isTemplateMode && __( 'Template' ) }
+				{ isPatternMode && __( 'Pattern' ) }
+				{ ! isTemplateMode && ! isPatternMode && documentLabel }
```

Comment:
I think the issue here is that `postType` is still 'post' when editing a pattern, and I think we need to try ensuring it's `wp_block` in the editor store. If the post type is set to the correct thing, `documentLabel` will have the right value.